commit 5b6a1098a98aeec51e5de0df7dc87c5b27c8c7b1
Author: Andrew Brampton <me@bramp.net>
Date:   Sun Nov 6 22:36:27 2011 -0500

    Populate the st_blocks and st_blksize fields of the stats struct.

diff --git a/libarchive/archive_entry_stat.c b/libarchive/archive_entry_stat.c
index ad772c9..91abab5 100644
--- a/libarchive/archive_entry_stat.c
+++ b/libarchive/archive_entry_stat.c
@@ -77,6 +77,8 @@ archive_entry_stat(struct archive_entry *entry)
 	st->st_rdev = archive_entry_rdev(entry);
 	st->st_size = archive_entry_size(entry);
 	st->st_mode = archive_entry_mode(entry);
+	st->st_blocks = (st->st_size + 511) / 512;
+	st->st_blksize = 4096;
 
 	/*
 	 * On systems that support high-res timestamps, copy that
